R/maic_anchored.R

```diff
@@ -131,62 +133,117 @@ maic_anchored <- function(weights_object,
+  retain_cols <- c("USUBJID", "ARM", outcome_cols, "weights")
+
   ipd <- ipd[, retain_cols, drop = FALSE]
   pseudo_ipd <- pseudo_ipd[, retain_cols, drop = FALSE]
-  dat <- rbind(ipd, pseudo_ipd)
+  dat <- rbind(ipd, pseudo_ipd) # only used if apply contrast method
   ipd$ARM <- factor(ipd$ARM, levels = c(trt_common, trt_ipd))
   pseudo_ipd$ARM <- factor(pseudo_ipd$ARM, levels = c(trt_common, trt_agd))
   dat$ARM <- factor(dat$ARM, levels = c(trt_common, trt_agd, trt_ipd))
 
   # ==> Inferential output ------------------------------------------
-  if (endpoint_type == "tte") {
-    # Analysis table (Cox model) before and after matching, incl Median Survival Time
-    # derive km w and w/o weights
-    kmobj_ipd <- survfit(Surv(TIME, EVENT) ~ ARM, ipd, conf.type = km_conf_type)
-    kmobj_ipd_adj <- survfit(Surv(TIME, EVENT) ~ ARM, ipd, weights = ipd$weights, conf.type = km_conf_type)
-    kmobj_agd <- survfit(Surv(TIME, EVENT) ~ ARM, pseudo_ipd, conf.type = km_conf_type)
-    res$descriptive[["survfit_ipd_before"]] <- survfit_makeup(kmobj_ipd)
-    res$descriptive[["survfit_ipd_after"]] <- survfit_makeup(kmobj_ipd_adj)
-    res$descriptive[["survfit_pseudo"]] <- survfit_makeup(kmobj_agd)
-    # derive median survival time
-    medSurv_ipd <- medSurv_makeup(kmobj_ipd, legend = "IPD, before matching", time_scale = time_scale)
-    medSurv_ipd_adj <- medSurv_makeup(kmobj_ipd_adj, legend = "IPD, after matching", time_scale = time_scale)
-    medSurv_agd <- medSurv_makeup(kmobj_agd, legend = "AgD, external", time_scale = time_scale)
-    medSurv_out <- rbind(medSurv_ipd, medSurv_ipd_adj, medSurv_agd)
-
-    res$inferential[["report_median_surv"]] <- medSurv_out
-
-    # fit PH Cox regression model
-    coxobj_ipd <- coxph(Surv(TIME, EVENT) ~ ARM, ipd, robust = TRUE)
-    coxobj_ipd_adj <- coxph(Surv(TIME, EVENT) ~ ARM, ipd, weights = weights, robust = TRUE)
-    coxobj_agd <- coxph(Surv(TIME, EVENT) ~ ARM, pseudo_ipd, robust = TRUE)
-
-    res$inferential[["ipd_coxph_before"]] <- coxobj_ipd
-    res$inferential[["ipd_coxph_after"]] <- coxobj_ipd_adj
-    res$inferential[["agd_coxph"]] <- coxobj_agd
-
-    # derive ipd exp arm vs agd exp arm via bucher
-    res_AC <- as.list(summary(coxobj_ipd_adj)$coef)[c(1, 4)]
-    res_BC <- as.list(summary(coxobj_agd)$coef)[c(1, 4)]
-    names(res_AC) <- names(res_BC) <- c("est", "se")
-    res_AB <- bucher(res_AC, res_BC, conf_lv = 0.95)
-    res_AB$est <- exp(res_AB$est)
-    res_AB$ci_l <- exp(res_AB$ci_l)
-    res_AB$ci_u <- exp(res_AB$ci_u)
-
-    res$inferential[["report_overall"]] <- rbind(
+  result <- if (endpoint_type == "tte") {
+    maic_anchored_tte(res, ipd, km_conf_type, pseudo_ipd, time_scale, weights_object, endpoint_name, trt_ipd, trt_agd)
+  } else {
+    stop("Endpoint type ", endpoint_type, " currently unsupported.")
+  }
+
+  result
+}
+
+
+# MAIC inference functions by outcome type ------------
+
+maic_anchored_tte <- function(res,
```

Comment:
@hoppanda I've factored this out into its own function. I think this will make it cleaner to implement for all the other outcome types. If you agree could you do the same for unanchored?

Reply:
Lets add the roxygen header but no @export and add @noRd